ixmp/reporting/utils.py

```diff
@@ -324,6 +326,18 @@ def to_series(self):
     def _constructor(self):
         return AttrSeries
 
+    def __finalize__(self, other, method=None, **kwargs):
+        """Propagate metadata from other to self.
+
+        This is identical to the version in pandas, except deepcopy() is added
+        so that the 'attrs' OrderedDict is not double-referenced.
+        """
+        if isinstance(other, NDFrame):
+            for name in self._metadata:
+                object.__setattr__(self, name,
+                                   deepcopy(getattr(other, name, None)))
```

Comment:
is it always needed to make a deep copy? may it have undesired consequences?

Reply:
Currently `AttrSeries.attrs` is a `collections.OrderedDict`, and the only element is '_unit', which is always a `pint` Unit object.

If we started storing things like references to `ixmp.Platform` objects there, it would become a problem. But I think that would indicate a deeper design issue…